xarray/core/dataset.py

```diff
@@ -2634,16 +2634,17 @@ def chunk(
         xarray.unify_chunks
         dask.array.from_array
         """
-        if chunks is None and chunks_kwargs is None:
+        if chunks is None and not chunks_kwargs:
             warnings.warn(
                 "None value for 'chunks' is deprecated. "
                 "It will raise an error in the future. Use instead '{}'",
                 category=FutureWarning,
             )
             chunks = {}
 
-        if not isinstance(chunks, Mapping):
-            # We need to ignore since mypy doesn't recognize this can't be `None`
+        if not isinstance(chunks, Mapping) and chunks is not None:
+            # The ignore seems to be required because of the type change (though not
+            # completely sure, the message is somewhat confusing)
             chunks = dict.fromkeys(self.dims, chunks)  # type: ignore[arg-type]
```

Comment:
Mypy doesn't like redefinitions. I've started to warm up to mypys opinion as well, it is annoying when a variable changes behavior significantly after a small if-check.  

Any time a variable changes type you should expect to create a new variable, so try something like this:
```python
        chunks_dict: dict
        if not isinstance(chunks, Mapping) and chunks is not None:
            chunks_dict = dict.fromkeys(self.dims, chunks)
```
Remember the downstream names as well.

Reply:
I agree! Though in this case, that didn't seem to completely satisfy mypy either. I think pushing until we remove the deprecation is probably reasonable...

Reply:
Aha, the error was about the .fromkeys second input. Renaming still seems to work, the typing for `chunks_mapping` can probably be improved.

Sidenote but I'm quite surprised this worked:
```python
dict.fromkeys((3,4,5), "auto")
Out[9]: {3: 'auto', 4: 'auto', 5: 'auto'}

# I was expecting:
{3: 'a', 4: 'u', 5: 't'}
```

Reply:
OK nice! I hadn't thought to make it `Mapping[Any, Any]`, I agree it's better now even if not perfect. Thanks!